src/pages/MerchantVoucherDashboardV2/VoucherTable.tsx

```diff
@@ -37,21 +43,32 @@ const VoucherTable = ({
   const [latestValue, setLatestValue] = useState<string>('');
 
   useEffect(() => {
+    // Reset error banner after selected row changes.
+    setErrorType(null);
+
     if (!editingRowGiftCardId) {
       setLatestValue('');
     }
-  }, [editingRowGiftCardId]);
+  }, [editingRowGiftCardId, setErrorType]);
 
   const onSave = useCallback(
     async (giftCardId: string) => {
-      const latestValueCents = parseFloat(latestValue) * 100;
-      await updateVoucher(giftCardId, latestValueCents);
-      setEditingRowGiftCardId(null);
-      fetchData();
-      // TODO: Show success banner.
-      // TODO: Show error banner on error.
+      try {
+        const latestValueCents = parseFloat(latestValue) * 100;
+        await updateVoucher(giftCardId, latestValueCents);
+        fetchData();
+        setEditingRowGiftCardId(null);
+        setErrorType(null);
+        setShowSuccessBanner(true);
+      } catch (error) {
+        if (error.message.indexOf('422') === -1) {
```

Comment:
you can use [`.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) now, no need to use `.indexOf()` anymore. But also, would it be better to check the status code itself instead of the message?

Reply:
just was worried a bit about browser compatibility for includes (not sure what browsers our merchants usually use). i can change it if we're not worried

when we throw the error from axios, we don't get a clean http error. it just gets captured somewhere in the message

Reply:
Gotchya on the axios stuff. And no need to worry about browser compatibility--the react builder takes care of that, we specify the browserlist in our package.json

Reply:
oh sweet, ok i changed it to includes then. much cleaner